offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/components/affiliation/update/UpdateAffiliationController.groovy

```diff
@@ -0,0 +1,21 @@
+package life.qbic.portal.offermanager.components.affiliation.update
+
+
+import life.qbic.business.persons.affiliation.Affiliation
+import life.qbic.business.persons.affiliation.update.UpdateAffiliationInput
+
+class UpdateAffiliationController {
+
+  private UpdateAffiliationInput useCaseInput
+
+  void updateAffiliation(Affiliation affiliation) {
+    if (useCaseInput == null) {
+      throw new RuntimeException("$this is missing the use case input.")
+    }
+    useCaseInput.updateAffiliation(affiliation)
+  }
+
+  void setUseCaseInput(UpdateAffiliationInput updateAffiliationInput) {
+    this.useCaseInput = updateAffiliationInput
+  }
```

Comment:
Why don't we just pass the use case as a parameter as we did before? Why should it be possible to initialize a controller without specifying the use case which it handles? The main disadvantage i see is that his introduces a new error source if someone forgets to set the useCaseInput